src/Rule/EmailHandler.php

```diff
@@ -29,7 +29,7 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
         if (!is_string($value)) {
             $valid = false;
         } elseif (!preg_match(
-            '/^(?P<name>(?:"?([^"]*)"?\s)?)(?:\s+)?((?P<open><?)((?P<local>.+)@(?P<domain>[^>]+))(?P<close>>?))$/i',
+            '/^(?P<name>(?:"?([^"]*)"?\s)?)(?:\s+)?((?P<open><?)((?P<local>.+)@(?P<domain>[^>]+))(?P<close>>?))/',
```

Comment:
What's the reason for removing anchor and case insensitivity?

Reply:
They both were escaped mutants. Couldn't find the relevant failing tests. Maybe you know such cases?

Reply:
Well, emails are case insensitive. Also it looks weird about removing the end anchor.

Reply:
As for the case, I added additional test case and it passes without the regex flag:

https://github.com/yiisoft/validator/blob/73a7cf099bbc323cf33f4290dcb3b71acde0c100/tests/Rule/EmailTest.php#L129

As for the end anchor, if it's essential, I can suggest restoring it and ignore specific mutant (and for this code block only if it's possible). All existing tests pass too though.

Reply:
I think it is essential.

Reply:
Seems, anchor does not affect because regex pattern don't contain case sensitive elements. We can removed him.

Reply:
Discussed in group chat, decided to leave it like that.